main/FatesInventoryInitMod.F90

```diff
@@ -374,6 +377,82 @@ subroutine initialize_sites_by_inventory(nsites,sites,bc_in)
 
          deallocate(patch_pointer_vec,patch_name_vec)
 
+         ! now that we've read in the patch and cohort info, check to see if there is any real age info
+         if ( (sites(s)%youngest_patch%age .eq. sites(s)%oldest_patch%age) .and. &
+              associated(sites(s)%youngest_patch%older) ) then
+
+            ! so there are at least two patches and the oldest and youngest are the same age.
+            ! this means that sorting by age wasn't very useful.  try sorting by total biomass instead
+
+            ! first calculate the biomass in each patch.  simplest way is to use the patch fusion criteria
+            currentpatch => sites(s)%youngest_patch
+            do while(associated(currentpatch))
+               call patch_pft_size_profile(currentPatch)
```

Comment:
Not that this has to happen in this PR. But this would be better as a function. It only updates a single array: currentPatch%pft_agb_profile(:,:)
It would be clearer from the calling function what it is changing.